src/main/scala/uk/gov/nationalarchives/consignmentexport/S3Files.scala

```diff
@@ -25,6 +26,7 @@ class S3Files(s3Utils: S3Utils)(implicit val logger: SelfAwareStructuredLogger[I
   def downloadFiles(files: List[ValidatedFileMetadata], bucket: String, consignmentId: UUID, consignmentReference: String, rootLocation: String): IO[Unit] = for {
     _ <- createDownloadDirectories(files, consignmentReference, rootLocation)
     _ <- files.filter(_.fileType != directoryType).map(file => {
+      IO.sleep(100 milliseconds)
       s3Utils.downloadFiles(bucket, s"$consignmentId/${file.fileId}", s"$rootLocation/$consignmentReference/${file.clientSideOriginalFilePath}".toPath.some)
```

Comment:
This won't run the sleep because you need to map or flatMap IO to get it to run.
There's a shortcut syntax so you can do `IO.sleep(100 milliseconds) >> s3Utils.downloadFiles...` or the more traditional `IO.sleep(100 milliseconds).flatMap(_ => s3Utils.downloadFiles...`

Reply:
This is what happens when you don't do any proper work for so long ...